test/integration/bastion/bastion_test.go

```diff
@@ -0,0 +1,628 @@
+
+	By("checking userData matches the constant")
+	//userdata ssh-public-key validation
+	Expect(string(*createdInstance.Metadata.Items[0].Value)).To(Equal(string(userDataConst)))
+}
+
+func verifyDeletion(ctx context.Context, project string, computeService *compute.Service, options *bastionctrl.Options) {
+	// bastion firewalls should be gone
+	// Check Firewall for Ingress / Egress
+	checkFirewallDoesNotExist(ctx, project, computeService, bastionctrl.FirewallIngressAllowSSHResourceName(options.BastionInstanceName))
+	checkFirewallDoesNotExist(ctx, project, computeService, bastionctrl.FirewallEgressAllowOnlyResourceName(options.BastionInstanceName))
+	checkFirewallDoesNotExist(ctx, project, computeService, bastionctrl.FirewallEgressDenyAllResourceName(options.BastionInstanceName))
+
+	// instance should be terminated and not found
+	_, err := computeService.Instances.Get(project, options.Zone, options.BastionInstanceName).Context(ctx).Do()
+	Expect(err).To(HaveOccurred())
+
+	// Disk should be terminated and not found
+	_, err = computeService.Disks.Get(project, options.Zone, bastionctrl.DiskResourceName(options.BastionInstanceName)).Context(ctx).Do()
+	Expect(err).To(HaveOccurred())
+}
+
+func checkFirewallDoesNotExist(ctx context.Context, project string, computeService *compute.Service, firewallName string) {
+	_, err := computeService.Firewalls.Get(project, firewallName).Context(ctx).Do()
+	Expect(err).To(HaveOccurred())
+}
+
+func checkFirewallExists(ctx context.Context, project string, computeService *compute.Service, firewallName string) {
+	firewall, err := computeService.Firewalls.Get(project, firewallName).Context(ctx).Do()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(firewall.Name).To(Equal(firewallName))
+}
+
+func namespace(name string) *corev1.Namespace {
+	return &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+}
+
+func setupEnvironmentObjects(ctx context.Context, c client.Client, namespace *corev1.Namespace, secret *corev1.Secret, cluster *extensionsv1alpha1.Cluster) {
+	Expect(c.Create(ctx, namespace)).To(Succeed())
+	Expect(c.Create(ctx, cluster)).To(Succeed())
+	Expect(c.Create(ctx, secret)).To(Succeed())
+}
+
+func teardownShootEnvironment(ctx context.Context, c client.Client, namespace *corev1.Namespace, secret *corev1.Secret, cluster *extensionsv1alpha1.Cluster) {
+	Expect(client.IgnoreNotFound(c.Delete(ctx, secret))).To(Succeed())
+	Expect(client.IgnoreNotFound(c.Delete(ctx, cluster))).To(Succeed())
+	Expect(client.IgnoreNotFound(c.Delete(ctx, namespace))).To(Succeed())
+}
+
+func getMyPublicIPWithMask() (string, error) {
+	resp, err := http.Get("https://api.ipify.org")
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	ip := net.ParseIP(string(body))
+	var mask net.IPMask
+	if ip.To4() != nil {
+		mask = net.CIDRMask(24, 32) // use a /32 net for IPv4
```

Comment:
```suggestion
		mask = net.CIDRMask(24, 32) // use a /24 net for IPv4
```
you missed to adapt the comment

Reply:
oh. fixed cf8c4e18